File: Labs/Lab 5/C++/Rows5/RowLib.cpp
#include "RowLib.h"
#include <algorithm>  // for std::count

int Numbers::getLength() const {
    return static_cast<int>(str.length() * 2);
}

std::string Numbers::increase() const {
    std::string res;
    for (char c : str) {
        res += c;
        res += c;
    }
    return res;
}

int Symbols::getLength() const {
    // кожен 'k' додає ще одну позицію
    return static_cast<int>(str.length() + std::count(str.begin(), str.end(), 'k'));
}

std::string Symbols::increase() const {
    std::string res;
    for (char c : str) {
        res += c;
        if (c == 'k') {
            res += 'k';
        }
    }
    return res;
}

File: Labs/Lab 5/C#/Row5/RowLib.cs
using System;
using System.Linq;
using System.Text;

namespace RowLib
{
    public abstract class Rows
    {
        protected string str;
        public Rows(string s) { str = s; }
        public abstract int GetLength();
        public abstract string Increase();
    }

    // дублювання кожного символу
    public class Numbers : Rows
    {
        public Numbers(string s) : base(s) { }
        public override int GetLength() => str.Length * 2;
        public override string Increase()
        {
            var sb = new StringBuilder();
            foreach (char c in str)
            {
                sb.Append(c).Append(c);
            }
            return sb.ToString();
        }
    }

    // дублювання лише символу 'k'
    public class Symbols : Rows
    {
        public Symbols(string s) : base(s) { }
        public override int GetLength() => str.Length + str.Count(c => c == 'k');
        public override string Increase()
        {
            var sb = new StringBuilder();
            foreach (char c in str)
            {
                sb.Append(c);
                if (c == 'k')
                    sb.Append('k');
            }
            return sb.ToString();
        }
    }
}